File: publishing-example-java-app/src/test/java/com/ably/tracking/example/javapublisher/AssetPublisherTest.java
package com.ably.tracking.example.javapublisher;

import com.ably.tracking.publisher.AssetPublisher;

import org.junit.Test;

public class AssetPublisherTest {
    @Test
    public void publishersFactoryShouldNotThrowError() {
        AssetPublisher.publishers();
    }

    @Test
    public void publishersShouldAllowToSetDeliveryDataWithOptionalParams() {
        AssetPublisher.publishers().delivery("some_id", null, null);
    }
}

File: publishing-example-app/src/test/java/com/ably/tracking/example/publisher/AssetPublisherTest.kt
package com.ably.tracking.example.publisher

import com.ably.tracking.publisher.AssetPublisher
import org.junit.Test

class AssetPublisherTest {
    @Test
    fun `publishers factory should not throw error`() {
        AssetPublisher.publishers()
    }

    @Test
    fun `publishers should allow to set delivery data with optional params`() {
        AssetPublisher.publishers().delivery("some_id")
    }
}
